src/utils/cpg.ts

```diff
@@ -0,0 +1,263 @@
+import { Action, ActionNode, Pathway } from 'pathways-model';
+import {
+  ActivityDefinition,
+  PlanDefinition,
+  PlanDefinitionAction,
+  Bundle,
+  BundleEntry,
+  Library
+} from 'fhir-objects';
+import shortid from 'shortid';
+import { isActionNode, findParents, isBranchNode } from './nodeUtils';
+import { R4 } from '@ahryman40k/ts-fhir-types';
+
+const LIBRARY_DRAFT = R4.LibraryStatusKind._draft;
+const PLANDEFINITION_DRAFT = R4.PlanDefinitionStatusKind._draft;
+const ACTIVITYDEFINITION_DRAFT = R4.ActivityDefinitionStatusKind._draft;
+const BUNDLE_COLLECTION = R4.BundleTypeKind._collection;
+const CONDITION_APPLICABILITY = R4.PlanDefinition_ConditionKindKind._applicability; // eslint-disable-line
+const EXPRESSION_CQL = R4.ExpressionLanguageKind._textCql;
+
+function createActivityDefinition(action: Action): ActivityDefinition {
+  const activityId = shortid.generate();
+  const kind =
+    action.resource.resourceType === 'Procedure' ? 'ServiceRequest' : action.resource.resourceType;
+
+  const activityDefinition: ActivityDefinition = {
+    id: activityId,
+    resourceType: 'ActivityDefinition',
+    meta: {
+      profile: ['http://hl7.org/fhir/uv/cpg/StructureDefinition/cpg-publishableactivity']
+    },
+    extension: [
+      {
+        url:
+          'http://build.fhir.org/ig/HL7/cqf-recommendations/StructureDefinition/cpg-knowledgeCapability.html',
+        valueCode: 'publishable'
+      }
+    ],
+    url: `urn:uuid:ActivityDefinition/${activityId}`,
+    version: '1.0',
+    name: activityId,
+    title: `ActivityDefinition: ${activityId}`,
+    status: ACTIVITYDEFINITION_DRAFT,
+    experimental: true,
+    date: Date.now().toString(),
```

Comment:
This is producing a unix timestamp (ex '1597238939135') which isn't a valid FHIR timestamp. I think `toISOString()` should work. Other than that everything in the sample bundle is POSTing correctly to our sample cqf-ruler server.